controllers/tasks.js

```diff
@@ -142,23 +143,48 @@ const updateTaskStatus = async (req, res, next) => {
   try {
     const taskId = req.params.id;
     const { dev } = req.query;
-    const { id: userId } = req.userData;
+    const { id: userId, username } = req.userData;
     const task = await tasks.fetchSelfTask(taskId, userId);
 
     if (task.taskNotFound) return res.boom.notFound("Task doesn't exist");
     if (task.notAssignedToYou) return res.boom.forbidden("This task is not assigned to you");
     if (task.taskData.status === "VERIFIED")
       return res.boom.forbidden("Status cannot be updated. Please contact admin.");
 
-    await tasks.updateTask(req.body, taskId);
+    const taskLog = {
+      type: "task",
+      meta: {
+        userId,
+        taskId,
+        username,
+      },
+      body: {
+        messages: [],
+      },
+    };
+
+    if (req.body.status) {
+      taskLog.body.messages.push(`${username} changed task status to ${req.body.status}`);
+      taskLog.body.newStatus = req.body.status;
+    }
+    if (req.body.percentCompleted) {
+      taskLog.body.messages.push(`${username} changed task percent Completed to ${req.body.percentCompleted}`);
+      taskLog.body.newPercentCompleted = req.body.percentCompleted;
```

Comment:
do we need to put username here as we are already sending the username in meta above